src/routes/Redirect.tsx

```diff
@@ -0,0 +1,22 @@
+import { useEffect } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+const Redirect = () => {
+  const [searchParams] = useSearchParams();
+
+  useEffect(() => {
+    const token = searchParams.get('token'); // 쿼리 파라미터에서 token 추출
+
+    if (token !== null && token.trim() !== '') {
+      // token이 null이 아니고 빈 문자열도 아닌 경우
+      localStorage.setItem('token', token); // 로컬 스토리지에 저장
+      window.location.href = '/'; // 메인 페이지로 리다이렉트
+    } else {
+      console.error('Token not found or is empty in URL');
+    }
+  }, [searchParams]);
```

Comment:
token을 먼저 꺼내두는 게 의존성배열 관리할 때 유리할 것 같습니다 👀 
별개로 메인페이지로 리다이렉트할 때 `window.location.href = '/'` 보다 `navigate('/')` 가 나을 수도 있을 것 같아요!

```suggestion
  const [searchParams] = useSearchParams();

  const token = searchParmas.get('token')?.trim();

  useEffect(() => {
    if (token !== null && token !== '') {
      // token이 null이 아니고 빈 문자열도 아닌 경우
      localStorage.setItem('token', token); // 로컬 스토리지에 저장
      window.location.href = '/'; // 메인 페이지로 리다이렉트
    } else {
      console.error('Token not found or is empty in URL');
    }
  }, [token]);
```

Reply:
넵 반영해서 window.location.href 대신 navigate로 수정하였습니다, 리뷰 감사합니다!